pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test: TestLogic_mixed_version_sequence_per_node_cache failed

pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test.TestLogic_mixed_version_sequence_per_node_cache [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14254518?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14254518?buildTab=artifacts#/) on master @ [583bfa4868b1aecedff86102ae55cde3b1748fd6](https://github.com/cockroachdb/cockroach/commits/583bfa4868b1aecedff86102ae55cde3b1748fd6):


```
=== RUN   TestLogic_mixed_version_sequence_per_node_cache
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/035e488ec2993725f1d7ba8eb7150687/logTestLogic_mixed_version_sequence_per_node_cache2381433756
    test_log_scope.go:81: use -show-logs to present logs inline
[23:30:08] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache: 6 statements
[23:30:40] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache: 10 statements
    logic.go:2952: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache:63: SELECT crdb_internal.is_at_least_version('23.2-18')
        expected:
            true
            
        but found (query options: "retry") :
            false
            
[23:31:26] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache: 12 statements
[23:31:26] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache with config cockroach-go-testserver-23.2: 12 tests, 1 failures
    logic.go:4130: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache:68: error while processing
    logic.go:4130: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8579/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-23.2/cockroach-go-testserver-23_2_test_/cockroach-go-testserver-23_2_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache:68: too many errors encountered, skipping the rest of the input
cockroach logs captured in: /artifacts/tmp/_tmp/035e488ec2993725f1d7ba8eb7150687/cockroach-logs3040208645
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/035e488ec2993725f1d7ba8eb7150687/logTestLogic_mixed_version_sequence_per_node_cache2381433756
--- FAIL: TestLogic_mixed_version_sequence_per_node_cache (91.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_sequence_per_node_cache.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36421

--- a/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache
+++ b/pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache
@@ -47,6 +47,9 @@ upgrade 1
 
 upgrade 2
 
+statement ok
+SET CLUSTER SETTING version = crdb_internal.node_executable_version();
+
 query T nodeidx=1
 SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
 ----

